### PR TITLE
Reduce NMBObjCMatcher usages

### PR DESCRIPTION
**Pull request by ikesyo**
See also #592.

### CODE REVIEW

File: Sources/Nimble/Matchers/BeLogical.swift [hunk 1]
```diff
@@ -141,10 +141,13 @@ extension NMBObjCMatcher {
...
     }
 
-    @objc public class func beFalseMatcher() -> NMBObjCMatcher {
-        return NMBObjCMatcher(canMatchNil: false) { actualExpression, failureMessage in
-            let expr = actualExpression.cast { ($0 as? NSNumber)?.boolValue ?? false }
-            return try beFalse().matches(expr, failureMessage: failureMessage)
+    @objc public class func beFalseMatcher() -> NMBMatcher {
+        return NMBPredicate { actualExpression in
+            let expr = actualExpression.cast { value -> Bool? in
+                guard let value = value else { return nil }
```
Comment (ikesyo) [inline Sources/Nimble/Matchers/BeLogical.swift]: `nil` should not be converted to `false` here.

File: Tests/NimbleTests/objc/ObjCEndWithTest.m [hunk 2]
```diff
@@ -13,7 +13,6 @@ - (void)testPositiveMatches {
     expect(@"hello world!").toNot(endWith(@"hello"));
     expect(array).to(endWith(@2));
     expect(array).toNot(endWith(@1));
-    expect(@1).toNot(contain(@"foo"));
```
Comment (ikesyo) [inline Tests/NimbleTests/objc/ObjCEndWithTest.m]: Testing `contain` matcher in `ObjCEndWithTest.m` is odd (and the fact the test is passing is odd as well) 😛

File: Sources/Nimble/Matchers/BeLogical.swift (hunk 1 above)
Comment (sharplet) [reply Sources/Nimble/Matchers/BeLogical.swift]: Oh, good catch. Should we add a test that covers this difference between `beFalse()` and `beFalsy()`? We're also accepting values that fail to cast to `NSNumber`, should we also return nil if the cast fails?

Comment (ikesyo) [reply Sources/Nimble/Matchers/BeLogical.swift]: That is already covered here: https://github.com/Quick/Nimble/blob/7de081475abd9db7e43355990c8f8ec0c178efe3/Tests/NimbleTests/objc/ObjCBeFalseTest.m#L24-L29

Previously the behavior is ensured by `NMBObjCMatcher(canMatchNil: false)`: https://github.com/Quick/Nimble/blob/7de081475abd9db7e43355990c8f8ec0c178efe3/Sources/Nimble/Adapters/NMBObjCMatcher.swift#L47-L52

Comment (ikesyo) [reply Sources/Nimble/Matchers/BeLogical.swift]: What I wanted to say that is that the message

> `nil` should not be converted to `false` here.

is only applicable to `NMBPredicate` in this PR, but not to the previous `NMBObjCMatcher` impl.

**Comment by ikesyo** [comment]
I will go ahead 🚀 

Thanks for the review @sharplet!

**Pull request merged**